crates/telio-network-monitors/src/mac.rs

```diff
@@ -0,0 +1,103 @@
+//! Module to monitor network changes in Apple
+
+use network_framework_sys::{
+    nw_interface_get_name, nw_interface_t, nw_path_monitor_create, nw_path_monitor_set_queue,
+    nw_path_monitor_start, nw_path_monitor_t, nw_path_t,
+};
+use parking_lot::Mutex;
+use std::{
+    cell::RefCell,
+    ffi::{c_long, c_void, CStr},
+    rc::Rc,
+};
+use telio_utils::{telio_log_info, telio_log_warn};
+
+use crate::monitor::PATH_CHANGE_BROADCAST;
+
+/// Dispatch queue priority as high priority queue
+pub const DISPATCH_QUEUE_PRIORITY_HIGH: c_long = 2;
+/// Dispatch queue priority as default queue
+pub const DISPATCH_QUEUE_PRIORITY_DEFAULT: c_long = 0;
+/// Dispatch queue priority as low priority queue
+pub const DISPATCH_QUEUE_PRIORITY_LOW: c_long = -2;
+/// Dispatch queue priority as background queue
+pub const DISPATCH_QUEUE_PRIORITY_BACKGROUND: c_long = -1 << 15;
+
+/// Vector containing interface names in OS order
+pub static INTERFACE_NAMES_IN_OS_PREFERENCE_ORDER: Mutex<Vec<String>> = Mutex::new(Vec::new());
+
+extern "C" {
+    /// Obj-c signature:
+    /// void nw_path_monitor_set_update_handler(nw_path_monitor_t monitor, nw_path_monitor_update_handler_t update_handler);
+    /// typedef void (^nw_path_monitor_update_handler_t)(nw_path_t path);
+    pub fn nw_path_monitor_set_update_handler(
+        monitor: nw_path_monitor_t,
+        update_handler: *const c_void,
+    );
+    /// Obj-c signature:
+    /// void nw_path_enumerate_interfaces(nw_path_t path, nw_path_enumerate_interfaces_block_t enumerate_block);
+    /// typedef bool (^nw_path_enumerate_interfaces_block_t)(nw_interface_t interface);
+    pub fn nw_path_enumerate_interfaces(path: nw_path_t, enumerate_block: *const c_void);
+}
+
+/// Setup network path monitor in Apple framework
```

Comment:
Again the fn comments are not the useful. 

It's clear for the dev that this function does that, but for instance "why do we need an objective C block?" and "how does monitor is setup?". If useful you can link some website for references.

Reply:
also, shouldn't we state/mark it as unsafe?